### PR TITLE
[prometheus] Add podAntiAffinity option switch for longterm

### PR DESCRIPTION
**Pull request by ykav-flant**
## Description

Adds an option to switch podAntiAffinity rule type for longterm Prometheus instance.

## Why do we need it, and what problem does it solve?

* Resolves an issue where the Kubernetes scheduler places both the longterm and main Prometheus instances on the same node after a failure of one of the nodes, leading to resource contention. Even after the node is fixed, this setup persists, worsening performance. Switching podAntiAffinty to `Required` ensures longterm instance won't be scheduled on the same node as main.
* Fixes https://github.com/deckhouse/deckhouse/issues/7216.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: prometheus
type: feature
summary: Added `longtermPodAntiAffinity` options to module.
```

### CODE REVIEW

File: modules/300-prometheus/openapi/config-values.yaml [hunk 1]
```diff
@@ -288,6 +288,15 @@ properties:
...
       If the parameter is omitted or `false`, it will be determined [automatically](https://deckhouse.io/products/kubernetes-platform/documentation/v1/#advanced-scheduling).
+  longtermPodAntiAffinity:
+    type: string
+    default: "preferred"
+    description: |
+      Defines the podAntiAffinity configuration for the longterm Prometheus instance relative to the main Prometheus instance. The options are either `preferred` or `required`.
+      When set to `preferred`, the anti-affinity rule is soft, in some cases allowing scheduling longterm Prometheus instance alongside the main instance.
+      When set to `required`, the anti-affinity rule is strict, preventing the scheduling of the longterm Prometheus instance on the same node as the main instance.
+      If the parameter is omitted or set to false, the default behavior is `preferred`.
+    pattern: '^(preffered|required)?$'
```
Comment (nabokihms) [inline modules/300-prometheus/openapi/config-values.yaml:299]: enum: [Preferred, Required]

Comment (nabokihms) [inline modules/300-prometheus/openapi/config-values.yaml:299]:
```suggestion
    enum: [Preferred, Required]
```

File: modules/300-prometheus/openapi/config-values.yaml [hunk 2]
```diff
@@ -288,6 +288,15 @@ properties:
       The same as in the Pods' `spec.nodeSelector` parameter in Kubernetes.
 
       If the parameter is omitted or `false`, it will be determined [automatically](https://deckhouse.io/products/kubernetes-platform/documentation/v1/#advanced-scheduling).
+  longtermPodAntiAffinity:
+    type: string
+    default: "preferred"
```
Comment (nabokihms) [inline modules/300-prometheus/openapi/config-values.yaml:293]:
```suggestion
    default: "Preferred"
```

File: modules/300-prometheus/openapi/config-values.yaml (hunk 1 above)
Comment (ykav-flant) [reply modules/300-prometheus/openapi/config-values.yaml]: Fixed.

File: modules/300-prometheus/openapi/config-values.yaml (hunk 2 above)
Comment (ykav-flant) [reply modules/300-prometheus/openapi/config-values.yaml]: Fixed.

File: modules/300-prometheus/templates/prometheus/longterm/prometheus.yaml [hunk 3]
```diff
@@ -113,14 +113,23 @@ spec:
   {{- end }}
   affinity:
     podAntiAffinity:
-      preferredDuringSchedulingIgnoredDuringExecution:
-      - weight: 100
-        podAffinityTerm:
-          labelSelector:
+      {{- if eq .Values.prometheus.longtermPodAntiAffinity "required" }}
```
Comment (nabokihms) [inline modules/300-prometheus/templates/prometheus/longterm/prometheus.yaml:116]:
```suggestion
      {{- if eq .Values.prometheus.longtermPodAntiAffinity "Required" }}
```

Comment (ykav-flant) [reply modules/300-prometheus/templates/prometheus/longterm/prometheus.yaml]: Fixed

**Review by vladimirGuryanov — APPROVED** [review]

File: modules/300-prometheus/openapi/doc-ru-config-values.yaml [hunk 4]
```diff
@@ -169,6 +169,14 @@ properties:
       Структура, аналогичная `spec.nodeSelector` пода Kubernetes.
 
       Если ничего не указано или указано `false`, будет [использоваться автоматика](https://deckhouse.ru/products/kubernetes-platform/documentation/v1/#выделение-узлов-под-определенный-вид-нагрузки).
+  longtermPodAntiAffinity:
+    type: string
+    description: |
+      Устанавливает podAntiAffinity конфигурацию для longterm Prometheus по отношению к main Prometheus.
+      Доступные варианты: `Preferred` или `Required`.
+      `Preferred` правило позволит размещать longterm на одном узле с main, в случае необходимости.
+      `Required` правило предотвратит размещение longterm на одном узле с main, при любых условиях.
```
Comment (z9r5) [inline modules/300-prometheus/openapi/doc-ru-config-values.yaml:178]:
```suggestion
      Устанавливает конфигурацию podAntiAffinity для Prometheus longterm по отношению к Prometheus main.

      - `Preferred` — позволяет размещать Prometheus longterm на одном узле с Prometheus main, если нет возможности их разместить на разных узлах.
      - `Required` — не позволяет размещать Prometheus longterm на одном узле с Prometheus main ни при каких условиях.
      Если параметр не указан или установлен в `false`, по умолчанию используется вариант `Preferred`.
```

File: modules/300-prometheus/openapi/doc-ru-config-values.yaml [hunk 5]
```diff
@@ -169,6 +169,14 @@ properties:
...
+      Если параметр не указан или установлен в `false`, по умолчанию используется вариант `Preferred`.
```
Comment (z9r5) [inline modules/300-prometheus/openapi/doc-ru-config-values.yaml:179]:
```suggestion
      Если параметр не указан или установлен в `false`, по умолчанию используется вариант `Preferred`.
```
Он не может быть установлен в false же - в спеке enum?
Значение параметра по умолнчанию будет присутствовать на странице автоматически из спеки.

File: modules/300-prometheus/openapi/config-values.yaml [hunk 6]
```diff
@@ -288,6 +288,15 @@ properties:
...
 
       If the parameter is omitted or `false`, it will be determined [automatically](https://deckhouse.io/products/kubernetes-platform/documentation/v1/#advanced-scheduling).
+  longtermPodAntiAffinity:
+    type: string
+    default: "Preferred"
+    description: |
+      Defines the podAntiAffinity configuration for the longterm Prometheus instance relative to the main Prometheus instance. The options are either `Preferred` or `Required`.
+      When set to `Preferred`, the anti-affinity rule is soft, in some cases allowing scheduling longterm Prometheus instance alongside the main instance.
+      When set to `Required`, the anti-affinity rule is strict, preventing the scheduling of the longterm Prometheus instance on the same node as the main instance.
+      If the parameter is omitted or set to false, the default behavior is `Preferred`.
```
Comment (z9r5) [inline modules/300-prometheus/openapi/config-values.yaml:298]:
```suggestion
      Defines the podAntiAffinity configuration for the Prometheus longterm instance relative to the Prometheus main instance.

      - `Preferred` — the anti-affinity rule is soft, in some cases allowing scheduling Prometheus longterm instance alongside the Prometheus main instance.
      - `Required` — the anti-affinity rule is strict, preventing the scheduling of the Prometheus longterm instance on the same node as the Prometheus main instance.
```

File: modules/300-prometheus/openapi/doc-ru-config-values.yaml (hunk 5 above)
Comment (ykav-flant) [reply modules/300-prometheus/openapi/doc-ru-config-values.yaml:179]: Не может, из изначальной релизации затесалось.

File: modules/300-prometheus/openapi/doc-ru-config-values.yaml (hunk 4 above)
Comment (ykav-flant) [reply modules/300-prometheus/openapi/doc-ru-config-values.yaml:178]: Спасибо!

**Review by z9r5 — APPROVED** [review]

**Pull request closed without merging**